src/rt/shell.c

```diff
@@ -1111,7 +1111,8 @@ static char *shell_list_generator(const char *script, const char *text,
                                   int state, int prefix)
 {
    static Tcl_Obj *list = NULL;
-   static int index, len, max;
+   static int index, len;
+   static Tcl_Size max;
```

Comment:
```suggestion
#if TCL_MAJOR_VERSION >= 9
   static Tcl_Size max;
#else
   static int max;
#endif
```

There's no `Tcl_Size` in TCL 8.